src/main/java/seedu/address/logic/commands/EditCardCommand.java

```diff
@@ -0,0 +1,176 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the card identified "
+            + "by the index number used in the last card listing. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_FRONT + "FRONT] "
+            + "[" + PREFIX_BACK + "BACK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_FRONT + " What is 1 + 1? " + PREFIX_BACK + " 2";
+
+    public static final String MESSAGE_EDIT_CARD_SUCCESS = "Edited Card: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_CARD = "This card already exists in the address book.";
+
+    private final Index index;
+    private final EditCardDescriptor editCardDescriptor;
+
+    private Card cardToEdit;
+    private Card editedCard;
+
+    /**
+     * @param index of the card in the filtered card list to edit
+     * @param editCardDescriptor details to edit the card with
+     */
+    public EditCardCommand(Index index, EditCardDescriptor editCardDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editCardDescriptor);
+
+        this.index = index;
+        this.editCardDescriptor = new EditCardDescriptor(editCardDescriptor);
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        try {
+            model.updateCard(cardToEdit, editedCard);
+        } catch (DuplicateCardException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_CARD);
+        } catch (CardNotFoundException pnfe) {
+            throw new AssertionError("The target card cannot be missing");
+        }
+        model.updateFilteredCardList(PREDICATE_SHOW_ALL_CARDS);
+        return new CommandResult(String.format(MESSAGE_EDIT_CARD_SUCCESS, editedCard));
+    }
+
+    @Override
+    protected void preprocessUndoableCommand() throws CommandException {
+        List<Card> lastShownList = model.getFilteredCardList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_CARD_DISPLAYED_INDEX);
+        }
+
+        cardToEdit = lastShownList.get(index.getZeroBased());
+        editedCard = createEditedCard(cardToEdit, editCardDescriptor);
+    }
+
+    /**
+     * Creates and returns a {@code Card} with the details of {@code cardToEdit}
+     * edited with {@code editCardDescriptor}.
+     */
+    private static Card createEditedCard(Card cardToEdit, EditCardDescriptor editCardDescriptor) {
+        assert cardToEdit != null;
+
+        String updatedFront = editCardDescriptor.getFront().orElse(cardToEdit.getFront());
+        String updatedBack = editCardDescriptor.getBack().orElse(cardToEdit.getBack());
+
+        return new Card(updatedFront, updatedBack);
```

Comment:
this is not enough anymore, the Card's ID would've changed. Please use `new Card(cardToEdit.getId(), ...)`